cql3/statements/create_keyspace_statement.cc

```diff
@@ -148,34 +148,37 @@ std::optional<sstring> check_restricted_replication_strategy(
+            {"org.apache.cassandra.locator.LocalStrategy", locator::replication_strategy_type::local}
+    };
+
     sstring replication_strategy = locator::abstract_replication_strategy::to_qualified_class_name(
         *attrs.get_replication_strategy_class());
-    auto& topology = qp.proxy().get_token_metadata_ptr()->get_topology();
-    // SimpleStrategy is not recommended in any setup which already has - or
-    // may have in the future - multiple racks or DCs. So depending on how
-    // protective we are configured, let's prevent it or allow with a warning:
-    if (replication_strategy == "org.apache.cassandra.locator.SimpleStrategy") {
-        switch(qp.db().get_config().restrict_replication_simplestrategy()) {
-        case db::tri_mode_restriction_t::mode::TRUE:
-            throw exceptions::configuration_exception(
-                "SimpleStrategy replication class is not recommended, and "
-                "forbidden by the current configuration. Please use "
-                "NetworkToplogyStrategy instead. You may also override this "
-                "restriction with the restrict_replication_simplestrategy=false "
-                "configuration option.");
-        case db::tri_mode_restriction_t::mode::WARN:
-            return format("SimpleStrategy replication class is not "
-                "recommended, but was used for keyspace {}. The "
-                "restrict_replication_simplestrategy configuration option "
-                "can be changed to silence this warning or make it into an error.",
-                keyspace);
-        case db::tri_mode_restriction_t::mode::FALSE:
-            // Scylla was configured to allow SimpleStrategy, but let's warn
-            // if it's used on a cluster which *already* has multiple DCs:
-            if (topology.get_datacenter_endpoints().size() > 1) {
-                return "Using SimpleStrategy in a multi-datacenter environment is not recommended.";
-            }
-            break;
+    if (const auto& strategy_mapping = rs_to_type.find(replication_strategy); strategy_mapping != rs_to_type.end()) {
+        auto selected_strategy = strategy_mapping->second;
+        auto rs_warn_list = qp.db().get_config().replication_strategy_warn_list();
+        auto rs_fail_list = qp.db().get_config().replication_strategy_fail_list();
+
+        if (auto present_on_fail_list = std::find(rs_fail_list.begin(), rs_fail_list.end(), selected_strategy); present_on_fail_list != rs_fail_list.end()) {
+            ++stats.replication_strategy_warn_list_violations;
+            throw exceptions::configuration_exception(format(
+                    "{} replication class is not recommended, "
+                    "and forbidden by the current configuration, but was used for keyspace {}. "
+                    "Consider using e.g. NetworkTopologyStrategy instead. You may also override this "
```

Comment:
If you want to mention NetworkToplogyStrategy as an example, I think you should check first that it is on the allowed list :-)

Reply:
Yes and no, because somebody could potentially place all 4 replication strategies on warn/fail list, which wouldn't make sense and would also render our response useless (?) There's no good answer here, I'm afraid, if we don't want to over-engineer this message and corresponding `if` statements, because if everything's forbidden (which is dumb in itself), we cannot suggest anything.
OTOH I don't see any reason not to always use NetworkTopology as an example. It's definitely better than SimpleStrategy, and local/everywhere strategies are kind-of hidden/implicit, and also not a best general fits.

Reply:
So don't recommend any strategy in the error message - just say that **this** strategy is forbidden. The user can go to the documentation and see what other strategies exist, and if there's a recommended one.
It's good that error messages are self-contained and self-explanatory, but not if the error messages can be wrong.

Reply:
OK, removed NetworkTopology suggestion.